src/sentry/static/sentry/app/components/projectSelector.jsx

```diff
@@ -269,32 +272,56 @@ class ProjectSelectorItem extends React.PureComponent {
   };
 
   render() {
-    const {project, multi, inputValue, isChecked} = this.props;
+    const {project, multi, inputValue, isChecked, organization} = this.props;
 
     return (
-      <GlobalSelectionHeaderRow
-        checked={isChecked}
-        onCheckClick={this.handleClick}
-        multi={multi}
-      >
-        <BadgeWrapper multi={multi}>
-          <IdBadgeMenuItem
-            project={project}
-            avatarSize={16}
-            displayName={<Highlight text={inputValue}>{project.slug}</Highlight>}
-            avatarProps={{consistentWidth: true}}
-          />
-        </BadgeWrapper>
-        {project.isBookmarked && <BookmarkIcon src="icon-star-small-filled" />}
-      </GlobalSelectionHeaderRow>
+      <BadgeAndActionsWrapper>
+        <GlobalSelectionHeaderRow
+          checked={isChecked}
+          onCheckClick={this.handleClick}
+          multi={multi}
+        >
+          <BadgeWrapper multi={multi}>
+            <IdBadgeMenuItem
+              project={project}
+              avatarSize={16}
+              displayName={<Highlight text={inputValue}>{project.slug}</Highlight>}
+              avatarProps={{consistentWidth: true}}
+            />
+          </BadgeWrapper>
+          {multi && (
```

Comment:
If a user doesn't have multiple-project selection why do they not get the settings/star buttons?

Reply:
honestly—this was just me being lazy because I hadn't checked if it worked. let me go back and see.

Reply:
Update here: I spiked on this and got pretty far, but I'm running into an issue where I have to add `withProjects` liberally in order for this to work, which then breaks a bunch of tests and sends me down the tubes. 

I think I'm gonna add an internal state to `BookmarkStar`, like we talked about below, which will make things easier in situations where connecting to the projectsStore isn't trivial.

Reply:
I got this working: all versions of the project dropdown now use `withProjects` and have the settings icon, bookmark star, and add icon.